serf/messages.go

```diff
@@ -136,6 +138,23 @@ func encodeMessage(t messageType, msg interface{}) ([]byte, error) {
 	return buf.Bytes(), err
 }
 
+// encodeRelayMessage wraps a message in the messageRelayType, adding the length and
+// address of the end recipient to the front of the message
+func encodeRelayMessage(t messageType, addr net.UDPAddr, msg interface{}) ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
```

Comment:
For future expansion it's probably better to use msgpack for the header, too, and just write a struct out here. You should be able to run one encoder and then the other to put the two messages back to back. Should work similarly on the read side as well.